File: frontend/src/api/news.ts
import api from './index'

export interface News {
  id: number
  title: string
  summary?: string
  content?: string
  source?: string
  sourceUrl?: string
  thumbnail?: string
  category: string
  viewCount: number
  isHot: boolean
  hotScore: number
  publishedAt: string
  createdAt: string
  updatedAt: string
}

export interface NewsRequest {
  title: string
  summary?: string
  content?: string
  source?: string
  sourceUrl?: string
  thumbnail?: string
  category?: string
}

export interface PaginatedResponse<T> {
  records: T[]
  total: number
  size: number
  current: number
  pages: number
}

export const newsApi = {
  // Create a new news item
  create: (data: NewsRequest) =>
    api.post<News>('/news', data),

  // Get news by ID
  getById: (id: number) =>
    api.get<News>(`/news/${id}`),

  // List all news with pagination
  list: (page: number = 0, size: number = 10) =>
    api.get<PaginatedResponse<News>>(`/news?page=${page}&size=${size}`),

  // List hot news
  listHot: (limit: number = 10) =>
    api.get<News[]>(`/news/hot?limit=${limit}`),

  // List news by category
  listByCategory: (category: string, page: number = 0, size: number = 10) =>
    api.get<PaginatedResponse<News>>(`/news/category/${category}?page=${page}&size=${size}`),

  // Search news
  search: (keyword: string, page: number = 0, size: number = 10) =>
    api.get<PaginatedResponse<News>>(`/news/search?keyword=${keyword}&page=${page}&size=${size}`),

  // Update news
  update: (id: number, data: Partial<NewsRequest & { isHot?: boolean; hotScore?: number }>) =>
    api.put<News>(`/news/${id}`, data),

  // Delete news
  delete: (id: number) =>
    api.delete(`/news/${id}`),

  // Set news as hot
  setHot: (id: number, isHot: boolean, hotScore: number) =>
    api.put<News>(`/news/${id}/hot`, { isHot, hotScore })
}

export default newsApi

File: src/main/java/com/aiproject/module/news/model/News.java
package com.aiproject.module.news.model;

import com.baomidou.mybatisplus.annotation.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * News Entity
 * Represents a hot news item
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("news")
public class News {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String title;

    private String summary;

    private String content;

    private String source;

    @TableField("source_url")
    private String sourceUrl;

    private String thumbnail;

    private NewsCategory category;

    @TableField("view_count")
    @Builder.Default
    private Long viewCount = 0L;

    @TableField("is_hot")
    @Builder.Default
    private Boolean isHot = false;

    @TableField("hot_score")
    @Builder.Default
    private Integer hotScore = 0;

    @TableField(value = "published_at")
    private LocalDateTime publishedAt;

    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(value = "updated_at", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    public enum NewsCategory {
        TECHNOLOGY,  // 科技
        FINANCE,     // 财经
        POLITICS,    // 政治
        SPORTS,      // 体育
        ENTERTAINMENT, // 娱乐
        HEALTH,      // 健康
        SCIENCE,     // 科学
        WORLD,       // 国际
        DOMESTIC,    // 国内
        OTHER        // 其他
    }
}
